samples/templates/express-direct-auth/web-server/routes/login.js

```diff
@@ -1,25 +1,67 @@
 const express = require('express');
 const { getAuthClient } = require('../utils');
+const sampleConfig = require('../../config').webServer;
 
 const router = express.Router();
 
-router.get('/login', (_, res) => {
+const renderLogin = (req, res) => {
   res.render('login');
+};
+
+const renderLoginWithWidget = (req, res) => {
+  const authClient = getAuthClient(req);
+  authClient.idx.interact()
+    .then(idxRes => {
+      const { 
+        interactionHandle, 
+        meta: { codeChallenge, codeChallengeMethod } 
+      } = idxRes;
+      const widgetConfig = JSON.stringify({
+        baseUrl: sampleConfig.oidc.issuer.split('/oauth2')[0],
+        clientId: sampleConfig.oidc.clientId,
+        redirectUri: sampleConfig.oidc.redirectUri,
+        authParams: {
+          issuer: sampleConfig.oidc.issuer,
+          scopes: sampleConfig.oidc.scope.split(' '),
+        },
+        useInteractionCodeFlow: true,
```

Comment:
setting `useInteractionCodeFlow` is not required when passing `interactionHandle`. However we should error here if `interactionHandle` does not exist. Otherwise, the widget will try to get a new interactionHandle from `/interact`.  I'm sure it's being rendered with `showSignInAndRedirect` but I think we want to set `redirect: "always"` explicitly in the options.